### PR TITLE
python_testing: Improve TC_IDM_10_1  error msgs

### PR DESCRIPTION
**Pull request by soares-sergio**
* Fix the prefix ID displayed in TC_IDM_10_1 error messages
* Add "(Test Vendor)" to the error message when the ID is in the test vendor range

Old error messages:
```
chip-all-clusters-app,"  Problem: ProblemSeverity.ERROR
  test_name: test_TC_IDM_10_1
  location:
  Endpoint: 1,
  Cluster:  80 (0x50) ModeSelect
  Attribute:4293984256 (0xfff10000)
  problem: Attribute with bad prefix 65531 in cluster 80"

chip-all-clusters-app,"  Problem: ProblemSeverity.ERROR
  test_name: test_TC_IDM_10_1
  location:
  Endpoint: 1,
  Cluster:  4293984256 (0xfff10000) Unknown cluster
  problem: Bad cluster id prefix 4293984256
  spec_location: Manufacturer Extensible Identifier (MEI)"
```

New error messages:
```
chip-all-clusters-app,"  Problem: ProblemSeverity.ERROR
  test_name: test_TC_IDM_10_1
  location:
  Endpoint: 1,
  Cluster:  80 (0x50) ModeSelect
  Attribute:4293984256 (0xfff10000)
  problem: Attribute with bad prefix 0xfff1 in cluster 80 (Test Vendor)"

chip-all-clusters-app,"  Problem: ProblemSeverity.ERROR
  test_name: test_TC_IDM_10_1
  location:
  Endpoint: 1,
  Cluster:  4293984256 (0xfff10000) Unknown cluster
  problem: Bad cluster id prefix 0xfff1 (Test Vendor)
  spec_location: Manufacturer Extensible Identifier (MEI)"
```

### CODE REVIEW

File: src/python_testing/TC_DeviceBasicComposition.py [hunk 1]
```diff
@@ -472,12 +476,12 @@ class RequiredMandatoryAttribute:
...
                     self.record_error(self.get_test_name(
-                    ), location=location, problem=f'Attribute with bad prefix {attribute_id} in cluster {cluster_id}', spec_location='Manufacturer Extensible Identifier (MEI)')
+                    ), location=location, problem=f'Attribute with bad prefix {hex(bad)[:6]} in cluster {cluster_id}' + (' (Test Vendor)' if is_test_vendor(bad) else ''), spec_location='Manufacturer Extensible Identifier (MEI)')
                     success = False
                 for bad in bad_cmds:
                     location = CommandPathLocation(endpoint_id=endpoint_id, cluster_id=cluster_id, command_id=bad)
                     self.record_error(self.get_test_name(
-                    ), location=location, problem=f'Command with bad prefix {attribute_id} in cluster {cluster_id}', spec_location='Manufacturer Extensible Identifier (MEI)')
+                    ), location=location, problem=f'Command with bad prefix {hex(bad)[:6]} in cluster {cluster_id}' + (' (Test Vendor)' if is_test_vendor(bad) else ''), spec_location='Manufacturer Extensible Identifier(MEI)')
```
Comment (tcarmelveilleux) [inline src/python_testing/TC_DeviceBasicComposition.py:484]:
```suggestion
                    ), location=location, problem=f'Command with bad prefix 0x{bad:04x} in cluster 0x{cluster_id:08x}' + (' (Test Vendor)' if is_test_vendor(bad) else ''), spec_location='Manufacturer Extensible Identifier(MEI)')
```

File: src/python_testing/TC_DeviceBasicComposition.py [hunk 2]
```diff
@@ -472,12 +476,12 @@ class RequiredMandatoryAttribute:
                 for bad in bad_attrs:
                     location = AttributePathLocation(endpoint_id=endpoint_id, cluster_id=cluster_id, attribute_id=bad)
                     self.record_error(self.get_test_name(
-                    ), location=location, problem=f'Attribute with bad prefix {attribute_id} in cluster {cluster_id}', spec_location='Manufacturer Extensible Identifier (MEI)')
+                    ), location=location, problem=f'Attribute with bad prefix {hex(bad)[:6]} in cluster {cluster_id}' + (' (Test Vendor)' if is_test_vendor(bad) else ''), spec_location='Manufacturer Extensible Identifier (MEI)')
```
Comment (tcarmelveilleux) [inline src/python_testing/TC_DeviceBasicComposition.py:479]:
```suggestion
                    ), location=location, problem=f'Attribute with bad prefix 0x{bad:04x} in cluster 0x{cluster_id:08x}' + (' (Test Vendor)' if is_test_vendor(bad) else ''), spec_location='Manufacturer Extensible Identifier (MEI)')
```

File: src/python_testing/TC_DeviceBasicComposition.py [hunk 3]
```diff
@@ -166,6 +166,10 @@ def check_no_duplicates(obj: Any) -> None:
         raise ValueError(f"Value {str(obj)} contains duplicate values")
 
 
+def is_test_vendor(id) -> bool:
+    return 0xFFF1_0000 <= id <= 0xFFF4_FFFF
```
Comment (tcarmelveilleux) [inline src/python_testing/TC_DeviceBasicComposition.py:170]:
```suggestion
def is_test_vendor(id: int) -> bool:
    vendor_id = (id >> 16) & 0xFFFF
    return 0xFFF1 <= vendor_id <= 0xFFF4
```

File: src/python_testing/TC_DeviceBasicComposition.py [hunk 4]
```diff
@@ -523,7 +527,7 @@ class RequiredMandatoryAttribute:
             for bad in bad_clusters_ids:
                 location = ClusterPathLocation(endpoint_id=endpoint_id, cluster_id=bad)
                 self.record_error(self.get_test_name(), location=location,
-                                  problem=f'Bad cluster id prefix {bad}', spec_location='Manufacturer Extensible Identifier (MEI)')
+                                  problem=f'Bad cluster id prefix {hex(bad)[:6]}' + (' (Test Vendor)' if is_test_vendor(bad) else ''), spec_location='Manufacturer Extensible Identifier (MEI)')
```
Comment (tcarmelveilleux) [inline src/python_testing/TC_DeviceBasicComposition.py:530]:
```suggestion
                                  problem=f'Bad cluster id prefix 0x{bad:04x}' + (' (Test Vendor)' if is_test_vendor(bad) else ''), spec_location='Manufacturer Extensible Identifier (MEI)')
```

File: src/python_testing/TC_DeviceBasicComposition.py (hunk 3 above)
Comment (cecille) [reply src/python_testing/TC_DeviceBasicComposition.py:170]: There are actually functions for this already - this test just happens to predate those and wasn't updated. Look in global_attribute_ids.py

Comment (soares-sergio) [reply src/python_testing/TC_DeviceBasicComposition.py]: Thanks, Cecille. I removed this function and used the checks in  global_attribute_ids.py.
I'll update the test and add checks for commands in a separate PR.

**Review by tcarmelveilleux — APPROVED** [review]

**Review by arkq — APPROVED** [review]

File: src/python_testing/TC_DeviceBasicComposition.py [hunk 5]
```diff
@@ -472,12 +474,12 @@ class RequiredMandatoryAttribute:
...
                     self.record_error(self.get_test_name(
-                    ), location=location, problem=f'Attribute with bad prefix {attribute_id} in cluster {cluster_id}', spec_location='Manufacturer Extensible Identifier (MEI)')
+                    ), location=location, problem=f'Attribute with bad prefix 0x{bad:04x} in cluster 0x{cluster_id:08x}' + (' (Test Vendor)' if attribute_id_type(bad) == AttributeIdType.kTest else ''), spec_location='Manufacturer Extensible Identifier (MEI)')
                     success = False
                 for bad in bad_cmds:
                     location = CommandPathLocation(endpoint_id=endpoint_id, cluster_id=cluster_id, command_id=bad)
                     self.record_error(self.get_test_name(
-                    ), location=location, problem=f'Command with bad prefix {attribute_id} in cluster {cluster_id}', spec_location='Manufacturer Extensible Identifier (MEI)')
+                    ), location=location, problem=f'Command with bad prefix 0x{bad:04x} in cluster 0x{cluster_id:08x}', spec_location='Manufacturer Extensible Identifier(MEI)')
```
[the remark this answers is not in the document]
Comment (tcarmelveilleux) [reply src/python_testing/TC_DeviceBasicComposition.py]: I prefer the the `0x:{value:04x}` pattern as there is less to remember and it's clear to readers what exactly will happen (and that the `0x` prefix is always there).

**Pull request merged**